ginivision/src/main/java/net/gini/android/vision/noresults/NoResultsFragmentHelper.java

```diff
@@ -0,0 +1,36 @@
+package net.gini.android.vision.noresults;
+
+import android.content.Context;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+
+import net.gini.android.vision.Document;
+import net.gini.android.vision.internal.ui.FragmentImplCallback;
+
+class NoResultsFragmentHelper {
+
+    private static final String ARGS_DOCUMENT = "GV_ARGS_DOCUMENT";
+
+    public static Bundle createArguments(@NonNull Document document) {
+        Bundle arguments = new Bundle();
+        arguments.putParcelable(ARGS_DOCUMENT, document);
+        return arguments;
+    }
+
+    static NoResultsFragmentImpl createFragmentImpl(@NonNull FragmentImplCallback fragment, @NonNull Bundle arguments) {
+        Document document = arguments.getParcelable(ARGS_DOCUMENT);
+        if (document != null) {
+            return new NoResultsFragmentImpl(fragment, document);
+        } else {
+            throw new IllegalStateException("NoResultsFragmentCompat and NoResultsFragmentStandard require a Document. Use the createInstance() method of these classes for instantiating.");
```

Comment:
Here as well. Maybe it is me that I'm wrong about try-catch on Java, but I would swear that Android Studio complained about it.

Reply:
The compiler complains about Exceptions, but RuntimeExceptions don't have to be handled. There are likely static code analysers that can warn about unhandled runtime exceptions, but haven't seen it in AS until now.

Reply:
Ok, I thought that runtime exception should be caught always! 👍